src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,71 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String DEFAULT_SUPPLY = "supply";
+    private static final String DEFAULT_BUY = "buy";
+    private static final String DEFAULT_RESULT = "result";
+    private static final String COMA = ",";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String string = readFromFile(fromFileName);
+        String result = resultOfDataReading(string);
+        recordToFile(result, toFileName);
+    }
+
+    private String readFromFile(String fromFileName) {
+        StringBuilder dataToString = new StringBuilder();
+        try {
+            BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName));
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                dataToString.append(line).append(COMA);
+            }
+            bufferedReader.close();
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read data from the file: " + fromFileName, e);
+        }
+        return dataToString.toString();
+    }
+
+    private String resultOfDataReading(String dataOfFile) {
```

Comment:
```suggestion
    private String generateReport(String fileData) {
```
method names should be verbs, save nouns for class names :)